Day0/bipul018/assignment2/second.c

```diff
@@ -0,0 +1,139 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+//Returns the position just after the number given
+char * readNum(char *,double *);
+
+//Flag = 1 means skip newline characters too
+char * skipWhites(char * p,int flag );
+int main(int argc, char *argv[])
+{
+
+    //Obtaining Source file
+    char filename[100]="expressions.txt";
+
+    //If given in command line, uses that file
+    if(argc>1)
+    {
+        strcpy(filename,argv[1]);
+    }
+    FILE *fp;
+    fp=fopen(filename,"r");
+    if(fp==NULL)
+    {
+        printf("File not found\n");
+        return 0;
+    }
+
+    fseek(fp, 0L, SEEK_END);
+  
+    // calculating the size of the file
+    long int fileSize = ftell(fp);
+    fseek(fp, 0L, SEEK_SET);
+
+    //Allocating memory to store the file
+    char* fileBuffer = (char*)malloc(fileSize);
+    memset(fileBuffer, 0, fileSize);
+    int size=0;
+    
+    while((fileBuffer[size]=fgetc(fp))!=EOF)
```

Comment:
there's fread function which will read entire contents at once given the size of the content, reading each character one by one might be slow